ServerCore/Pages/EventSpecific/PH20/PuzzlePicker.cshtml.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.EventSpecific.PH20
+{
+    /// <summary>
+    /// Page for choosing between different whistle stop options. This is done in the context
+    /// of a "choice" puzzle so that this page can be locked and unlocked
+    /// </summary>
+    [Authorize(Policy = "PlayerCanSeePuzzle")]
+    public class PuzzlePickerModel : EventSpecificPageModel
+    {
+        public PuzzlePickerModel(PuzzleServerContext serverContext, UserManager<IdentityUser> manager) : base(serverContext, manager)
+        {
+        }
+
+        public List<Puzzle> PuzzleOptions { get; set; }
+        public int PrereqPuzzleId { get; set; }
+
+        public async Task<IActionResult> OnGet(int puzzleId)
+        {
+            PrereqPuzzleId = puzzleId;
+
+            Puzzle puzzle = await (from puzz in _context.Puzzles
+                                   where puzz.ID == puzzleId
+                                   select puzz).FirstOrDefaultAsync();
+            if (puzzle == null)
+            {
+                return NotFound();
+            }
+
+            // Restrict this page to whistle stops
+            if (puzzle.MinutesOfEventLockout == 0)
+            {
+                return NotFound();
+            }
+
+            Team team = await UserEventHelper.GetTeamForPlayer(_context, Event, LoggedInUser);
+            var puzzleStateQuery = PuzzleStateHelper.GetFullReadOnlyQuery(_context, Event, null, team);
```

Comment:
pass "puzzle" for param 3 and then you do not need the where clause below.

Reply:
oh - I see you reuse this query below and that's why you did it like this. Well, you can certainly remove the team filter from the where clause since you specify it here.